### ISSUE TITLE
git-undo uses a common "help" flag for hard resetting

### ISSUE THREAD

**Issue by blackrobot** [tj/git-extras#695]
Some users (like me) assume that `-h` will display the usage documentation, just as calling `git <command> -h` does in other instances.

I propose the short `-h` option of `git undo` should be removed, and instead the user must use the long `--hard` form.

Alternatively, `git undo -h` could prompt for confirmation and `git undo -hf` could bypass the confirmation if the user wants to type less.

```bash
# displays the usage output...
$ git undo -h
usage: git undo [<commitcount>] [--soft | --hard]
# etc...

# or displays a confirmation
$ git undo -h
This will erase any changes since your last commit. Do you want to continue? [yN] 

# which the user can skip
$ git undo -h -f
HEAD is now at a1b2c3d4 ...
```

_I just had this happen to me, luckily I was able to recover my work with `git reset HEAD@{1}`._

**Comment by spacewander** [comment]
@blackrobot 
Sorry for the impact.
I just push a commit to master branch, add a confirmation when `-h` is specified.

**Comment by spacewander** [comment]
Closed as commit 16591d3788cbf27f1bba9192638211ba2454ae9f is pushed to master.
